notesapi/v1/tests/test_views.py

```diff
@@ -278,17 +289,34 @@ def test_update(self):
         """
         Ensure we can update an existing annotation.
         """
-        data = self._create_annotation(text=u"Foo")
-        payload = self.payload.copy()
-        payload.update({'id': data['id'], 'text': 'Bar'})
-        payload.update(self.headers)
-        url = reverse('api:v1:annotations_detail', kwargs={'annotation_id': data['id']})
-        response = self.client.put(url, payload, format='json')
-        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        def do_update(data, updated_fields):
+            """ Helper method for performing the update """
+            payload = self.payload.copy()
+            payload.update(updated_fields)
+            payload.update(self.headers)
+            url = reverse('api:v1:annotations_detail', kwargs={'annotation_id': data['id']})
+            response = self.client.put(url, payload, format='json')
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+            return response, self._get_annotation(data['id'])
 
-        annotation = self._get_annotation(data['id'])
+        data = self._create_annotation(text=u"Foo")
+        response, annotation = do_update(data, {'id': data['id'], 'text': 'Bar', 'tags': ["new", "tags"]})
```

Comment:
I think that these tags test cases should either be split up into unique test methods, or ddt'd together into one tags-related test method. Either way, I think we should have methods dedicated to testing different aspects of an update, rather than one catch-all method.